src/NATS.Client.KeyValueStore/NatsKVStore.cs

```diff
@@ -315,11 +315,27 @@ public async IAsyncEnumerable<NatsKVEntry<T>> WatchAsync<T>(string key, INatsDes
     {
         await using var watcher = await WatchInternalAsync<T>(key, serializer, opts, cancellationToken);
 
+        if (watcher.InitialConsumer.Info.NumPending == 0 && opts?.EndOfCurrentData != null)
+        {
+            if (await opts.EndOfCurrentData(cancellationToken))
+            {
+                yield break;
+            }
+        }
```

Comment:
This is the part that I agree we need an option for, and it may even be good to have a consistent option in `NatsKVOpts` and `NatsConsumeOpts`

Reply:
unfortunately this 'trick' only applies to push consumers which are only used for KV. for other consumers solution is to use fetch-no-wait. pull consumers would have a race between the consumer creation and pull request i.e. you might think there is data but by the time pull request is issued that might not be the case anymore causing enumeration to hang.